src/storm-dft/transformations/SftToBddTransformator.h

```diff
@@ -0,0 +1,260 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "storm-dft/storage/SylvanBddManager.h"
+#include "storm-dft/storage/dft/DFT.h"
+#include "storm-dft/utility/RelevantEvents.h"
+#include "storm/exceptions/NotSupportedException.h"
+#include "storm/utility/bitoperations.h"
+
+namespace storm {
+namespace transformations {
+namespace dft {
+
+/**
+ * Transformator for DFT -> BDD.
+ */
+template <typename ValueType>
+class SftToBddTransformator {
+   public:
+    using Bdd = sylvan::Bdd;
+
+    SftToBddTransformator(
+        std::shared_ptr<storm::storage::DFT<ValueType>> dft,
+        std::shared_ptr<storm::storage::SylvanBddManager> sylvanBddManager =
+            std::make_shared<storm::storage::SylvanBddManager>(),
+        storm::utility::RelevantEvents relevantEvents = {})
+        : dft{std::move(dft)},
+          sylvanBddManager{std::move(sylvanBddManager)},
+          relevantEvents{relevantEvents} {
+        // create Variables for the BEs
+        for (auto const& i : this->dft->getBasicElements()) {
+            // Filter constantBeTrigger
+            if (i->name() != "constantBeTrigger") {
+                variables.push_back(
+                    this->sylvanBddManager->createVariable(i->name()));
+            }
+        }
+    }
+
+    /**
+     * Transform a sft into a BDD
+     * representing the function of the toplevel gate.
+     *
+     * \exception storm::exceptions::NotSupportedException
+     * The given DFT is not a SFT
+     *
+     * \return The translated Bdd of the toplevel gate
+     */
+    Bdd const& transformTopLevel() {
+        auto const tlName{dft->getTopLevelGate()->name()};
+        if (relevantEventBdds.empty()) {
+            relevantEventBdds[tlName] =
+                translate(std::static_pointer_cast<
+                          storm::storage::DFTElement<ValueType> const>(
+                    dft->getTopLevelGate()));
+        }
+        // else relevantEventBdds is not empty and we maintain the invaraint
+        // that the toplevel gate is in there
```

Comment:
I can do that but it would fail in `relevantEventBdds[tlName]` regardless.
The only advantage would be a better error msg but only if storm is build in debug mode.
But then a simple backtrace would lead to this line and the comment above anyway.

Reply:
I am in favour of using the assert for a better error message (in debug mode). It also makes explicit that we are maintaining an invariant.
If I read the documentation correctly, the access via `[]` will also not throw an error but just inserts a default constructed object.

Reply:
Yea I somehow thought operator[] would behave like at() on maps. I don't know why though.
I would have added the assert in either way btw.